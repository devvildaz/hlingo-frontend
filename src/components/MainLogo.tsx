import { Center, Image, Text } from 'native-base';

const sizes = {
  sm: {
    logo: '36',
    text: '3xl',
  },
  md: {
    logo: '48',
    text: '4xl',
  },
  lg: {
    logo: '64',
    text: '5xl',
  },
};

interface Props {
  size?: 'sm' | 'md' | 'lg';
}

const MainLogo = ({ size = 'md' }: Props) => {
  return (
    <Center p={4}>
      <Image
        source={require('@assets/icon.png')}
        alt="Hololingo logo"
        size={sizes[size].logo}
        mb={2}
        accessibilityRole="image"
      />
      <Text bold fontSize={sizes[size].text}>
        Hololingo
      </Text>
    </Center>
  );
};

export default MainLogo;
