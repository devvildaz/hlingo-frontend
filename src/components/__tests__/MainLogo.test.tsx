import React from 'react';
import renderer from 'react-test-renderer';

import MainLogo from '@components/MainLogo';
import Wrapper from '../ProviderWrapper';

describe('<MainLogo />', () => {
  beforeAll(() => {
    jest.useFakeTimers();
  });

  it('should match the snapshot', () => {
    const component = renderer.create(
      <Wrapper>
        <MainLogo />
      </Wrapper>
    );

    let tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
