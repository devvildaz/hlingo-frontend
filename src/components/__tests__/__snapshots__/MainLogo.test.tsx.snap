// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainLogo /> should match the snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    dataSet={Object {}}
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "justifyContent": "center",
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 16,
      }
    }
  >
    <Image
      accessibilityLabel="Hololingo logo"
      alt="Hololingo logo"
      dataSet={Object {}}
      onError={[Function]}
      source={1}
      style={
        Object {
          "height": 192,
          "marginBottom": 8,
          "maxWidth": "100%",
          "width": 192,
        }
      }
    />
    <Text
      dataSet={Object {}}
      style={
        Object {
          "backgroundColor": undefined,
          "color": "#171717",
          "fontFamily": undefined,
          "fontSize": 36,
          "fontStyle": "normal",
          "fontWeight": "700",
          "letterSpacing": 0,
          "lineHeight": 54,
          "textDecorationLine": undefined,
        }
      }
    >
      Hololingo
    </Text>
  </View>
</RNCSafeAreaProvider>
`;
